src/Surfnet/ServiceProviderDashboard/Domain/Entity/ManageEntity.php

```diff
@@ -87,15 +89,17 @@ class ManageEntity
      */
     public static function fromApiResponse($data)
     {
-        $manageProtocol = isset($data['type']) ? $data['type'] : '';
+        $manageProtocol = self::extractManageProtocol($data);
 
         $attributeList = AttributeList::fromApiResponse($data);
         $metaData = MetaData::fromApiResponse($data);
         $oidcClient = null;
         if ($manageProtocol === Protocol::OAUTH20_RS) {
-            $oidcClient = OidcngResourceServerClient::fromApiResponse($data, $manageProtocol);
+            $oidcClient = OidcngResourceServerClient::fromApiResponse($data);
```

Comment:
A switch would be more readable here, as well as faster.

Reply:
Agreed